CMakeLists.txt

```diff
@@ -1,47 +1,60 @@
-project(ChimeraTK-DeviceAccess-EPICS-Backend)
-
 cmake_minimum_required(VERSION 3.16)
 
+project(ChimeraTK-DeviceAccess-EPICS-Backend)
+
 set(${PROJECT_NAME}_MAJOR_VERSION 00)
 set(${PROJECT_NAME}_MINOR_VERSION 02)
 set(${PROJECT_NAME}_PATCH_VERSION 01)
 include(cmake/set_version_numbers.cmake)
 
-option(BUILD_TESTS "Build tests." OFF)
-
 include(cmake/set_default_flags.cmake)
 include(cmake/set_default_build_to_release.cmake)
 include(cmake/enable_code_style_check.cmake)
 
 find_package(ChimeraTK-DeviceAccess 03.14 REQUIRED)
 
-include(FindPkgConfig)
+# Change project name, based on EPICS version
+OPTION(EPICS_VERSION "Select major version of EPICS to build (3 or 7)")
+
+if(NOT EPICS_VERSION)
+  set(EPICS_VERSION "7")
+endif()
 
-if(DEFINED EPICS_DIR)
-  set(ENV{PKG_CONFIG_PATH} $ENV{PKG_CONFIG_PATH}:${EPICS_DIR}/base/lib/pkgconfig)
+if(NOT EPICS_VERSION STREQUAL "3" AND NOT EPICS_VERSION STREQUAL "7")
+  message(FATAL_ERROR "Unknown EPICS version specified: ${EPICS_VERSION}!\n"
+    "Please pass to cmake command line one of the following:\n"
+    "-DEPICS_VERSION=3 to build against EPICS ver. 3.15\n"
+    "-DEPICS_VERSION=7 to build against EPICS ver. 7.x")
 endif()
 
-message("Using PKG_CONFIG_PATH=$ENV{PKG_CONFIG_PATH}")
-pkg_check_modules(EPICS REQUIRED epics-base-linux-x86_64)
-#pkg_get_variable(EPICS_LIBS epics-base-linux-x86_64 EPICS_BASE_IOC_LIBS)
-pkg_get_variable(EPICS_LIBS epics-base-linux-x86_64 EPICS_BASE_HOST_LIBS)
-pkg_get_variable(EPICS_ARCH epics-base-linux-x86_64 ARCH)
-pkg_get_variable(EPICS_PATH epics-base-linux-x86_64 FINAL_LOCATION)
+find_package(ChimeraTK-EPICS${EPICS_VERSION} REQUIRED)
+
+get_target_property(EPICS_BASE ChimeraTK::EPICS${EPICS_VERSION} INTERFACE_EPICS_BASE)
+get_target_property(EPICS_ARCH ChimeraTK::EPICS${EPICS_VERSION} INTERFACE_EPICS_ARCH)
 
 list(APPEND CMAKE_INSTALL_RPATH ${EPICS_LIBRARY_DIRS})
 include_directories(include ${EPICS_INCLUDE_DIRS})
 
 aux_source_directory(${CMAKE_SOURCE_DIR}/src library_sources)
 
-add_library(${PROJECT_NAME} SHARED ${library_sources} )
+add_library(${PROJECT_NAME} SHARED ${library_sources})
 set_target_properties(${PROJECT_NAME} PROPERTIES INSTALL_RPATH_USE_LINK_PATH TRUE)
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${${PROJECT_NAME}_FULL_LIBRARY_VERSION} SOVERSION ${${PROJECT_NAME}_SOVERSION})
 target_link_libraries(${PROJECT_NAME} PUBLIC ChimeraTK::ChimeraTK-DeviceAccess
-                                      PRIVATE ${EPICS_LIBS}
-                                      PRIVATE ${EPICS_LDFLAGS})
-if(BUILD_TESTS)
-  add_subdirectory(test)
-endif(BUILD_TESTS)
+  PRIVATE ChimeraTK::EPICS${EPICS_VERSION}
+)
+
+FIND_PACKAGE(Boost 1.83 COMPONENTS unit_test_framework)
+
+if(Boost_FOUND)
+  option(BUILD_TESTS "Build tests." ON)
```

Comment:
done

Reply:
Ah, this was a trick since we do not have BOOST 1.83 on Ubuntu 20 but the test does not compile with older BOOST versions... I should have written a comment, if I even forget this between implementation and review...